src/types.rs

```diff
@@ -636,5 +694,81 @@ mod tests {
         );
+        assert!(obj.subproblems.is_none());
+    }
+
+    // https://www.rfc-editor.org/rfc/rfc8555#section-6.7.1
+    #[test]
+    fn subproblems() {
+        const PROBLEM: &str = r#"{
+            "type": "urn:ietf:params:acme:error:malformed",
+            "detail": "Some of the identifiers requested were rejected",
+            "subproblems": [
+                {
+                    "type": "urn:ietf:params:acme:error:malformed",
+                    "detail": "Invalid underscore in DNS name \"_example.org\"",
+                    "identifier": {
+                        "type": "dns",
+                        "value": "_example.org"
+                    }
+                },
+                {
+                    "type": "urn:ietf:params:acme:error:rejectedIdentifier",
+                    "detail": "This CA will not issue for \"example.net\"",
+                    "identifier": {
+                        "type": "dns",
+                        "value": "example.net"
+                    }
+                }
+            ]
+        }"#;
+
+        let obj = serde_json::from_str::<Problem>(PROBLEM).unwrap();
+        assert_eq!(
+            obj.r#type,
+            Some("urn:ietf:params:acme:error:malformed".into())
+        );
+        assert_eq!(
+            obj.detail,
+            Some("Some of the identifiers requested were rejected".into())
+        );
+
+        let subproblems = obj.subproblems.as_ref().unwrap();
+        assert_eq!(subproblems.len(), 2);
+
+        let first_subproblem = subproblems.first().unwrap();
+        assert_eq!(
+            first_subproblem.identifier,
+            Some(Identifier::Dns("_example.org".into()))
+        );
+        assert_eq!(
+            first_subproblem.r#type,
+            Some("urn:ietf:params:acme:error:malformed".into())
+        );
+        assert_eq!(
+            first_subproblem.detail,
+            Some(r#"Invalid underscore in DNS name "_example.org""#.into())
+        );
+
+        let second_subproblem = subproblems.get(1).unwrap();
+        assert_eq!(
+            second_subproblem.identifier,
+            Some(Identifier::Dns("example.net".into()))
+        );
+        assert_eq!(
+            second_subproblem.r#type,
+            Some("urn:ietf:params:acme:error:rejectedIdentifier".into())
+        );
+        assert_eq!(
+            second_subproblem.detail,
+            Some(r#"This CA will not issue for "example.net""#.into())
+        );
+
+        let expected_display = "\
+    API error: Some of the identifiers requested were rejected (urn:ietf:params:acme:error:malformed): \
+    2 subproblems: \
+    error for _example.org: Invalid underscore in DNS name \"_example.org\" (urn:ietf:params:acme:error:malformed), \
+    error for example.net: This CA will not issue for \"example.net\" (urn:ietf:params:acme:error:rejectedIdentifier)";
+        assert_eq!(format!("{obj}"), expected_display);
```

Comment:
I'm generally nervous about writing brittle tests based on `Display` impls, but I also wanted to show what a problem with sub-problems renders like. 

WDYT? Keep this or axe it?

Reply:
I don't expect us to change this all that often and it is useful for human reviewers, so let's try keeping it for now.